### PR TITLE
Cloudformation getatt for Firehose and ES Domain

### PR DESCRIPTION
**Pull request by erka**
Added missed cloudformation getatt for KinesisFirehose::DeliveryStream and Elasticsearch::Domain

### CODE REVIEW

File: localstack/services/cloudformation/cloudformation_starter.py [hunk 1]
```diff
@@ -575,6 +575,26 @@ def SNS_Topic_get_cfn_attribute(self, attribute_name):
     SNS_Topic_get_cfn_attribute_orig = sns_models.Topic.get_cfn_attribute
     sns_models.Topic.get_cfn_attribute = SNS_Topic_get_cfn_attribute
 
+    # Patch ES get_cfn_attribute(..) method
+    def ES_get_cfn_attribute(self, attribute_name):
+        if attribute_name in ['Arn', 'DomainArn']:
+            return aws_stack.es_domain_arn(self.params.get('DomainName'))
+        if attribute_name == 'DomainEndpoint':
+            es_details = aws_stack.connect_to_service('es').describe_elasticsearch_domain(
```
Comment (whummer) [inline localstack/services/cloudformation/cloudformation_starter.py:583]: nit: Can we try to cache this value, to improve the performance and avoid that it gets requested from the API multiple times? Something like:
```
        if attribute_name == 'DomainEndpoint':
            if not hasattr(self, '_domain_endpoint'):
                es_details = aws_stack.connect_to_service('es').describe_elasticsearch_domain(
                    DomainName=self.params.get('DomainName'))
                self._domain_endpoint = es_details['DomainStatus']['Endpoint']
            return self._domain_endpoint
        ...
```

File: localstack/services/cloudformation/cloudformation_starter.py [hunk 2]
```diff
@@ -575,6 +575,26 @@ def SNS_Topic_get_cfn_attribute(self, attribute_name):
...
+                DomainName=self.params.get('DomainName'))
+            return es_details['DomainStatus']['Endpoint']
+        raise UnformattedGetAttTemplateException()
+
+    service_models.ElasticsearchDomain.get_cfn_attribute = ES_get_cfn_attribute
+
+    # Patch Firehose get_cfn_attribute(..) method
+    def Firehose_get_cfn_attribute(self, attribute_name):
+        if attribute_name == 'Arn':
+            return aws_stack.firehose_stream_arn(self.params.get('DeliveryStreamName'))
```
Comment (whummer) [inline localstack/services/cloudformation/cloudformation_starter.py:593]: Same as above - we could try to cache the value here, for performance reasons.

File: tests/integration/test_cloudformation.py [hunk 3]
```diff
@@ -735,6 +749,13 @@ def test_cfn_handle_elasticsearch_domain(self):
         tags = es_client.list_tags(ARN=status['ARN'])['TagList']
         self.assertEqual([{'Key': 'k1', 'Value': 'v1'}, {'Key': 'k2', 'Value': 'v2'}], tags)
 
+        for o in outputs:
+            if o['OutputKey'] in ['MyElasticsearchArn', 'MyElasticsearchDomainArn']:
+                self.assertEqual(o['OutputValue'], status['ARN'])
+            elif o['OutputKey'] == 'MyElasticsearchDomainEndpoint':
+                self.assertEqual(o['OutputValue'], status['Endpoint'])
+            else:
+                self.fail('unknown output')
```
Comment (whummer) [inline tests/integration/test_cloudformation.py:758]: nitpick: we could print the unexpected output here, for debugging purposes:
```
else:
    self.fail('Unexpected output: %s' % o)
```

File: localstack/services/cloudformation/cloudformation_starter.py (hunk 2 above)
Comment (erka) [reply localstack/services/cloudformation/cloudformation_starter.py:596]: Are you sure about this one @whummer? I believe `firehose_stream_arn` doesn't make any hard work.

File: localstack/services/cloudformation/cloudformation_starter.py [hunk 4]
```diff
@@ -575,6 +575,29 @@ def SNS_Topic_get_cfn_attribute(self, attribute_name):
     SNS_Topic_get_cfn_attribute_orig = sns_models.Topic.get_cfn_attribute
     sns_models.Topic.get_cfn_attribute = SNS_Topic_get_cfn_attribute
 
+    # Patch ES get_cfn_attribute(..) method
+    def ES_get_cfn_attribute(self, attribute_name):
+        if attribute_name in ['Arn', 'DomainArn']:
+            return aws_stack.es_domain_arn(self.params.get('DomainName'))
+        if attribute_name == 'DomainEndpoint':
+            if attribute_name == 'DomainEndpoint':
```
Comment (whummer) [inline localstack/services/cloudformation/cloudformation_starter.py:583]: nitpick: Looks like we have a duplicate check `if attribute_name == 'DomainEndpoint':` here

File: localstack/services/cloudformation/cloudformation_starter.py (hunk 2 above)
Comment (whummer) [reply localstack/services/cloudformation/cloudformation_starter.py:596]: Good point, agreed. 👍

File: localstack/services/cloudformation/cloudformation_starter.py (hunk 4 above)
Comment (erka) [reply localstack/services/cloudformation/cloudformation_starter.py]: I fixed it

**Comment by whummer** [comment]
Great, thanks for the quick turnaround.

**Pull request merged**